OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java

```diff
@@ -2169,20 +2169,16 @@ static void applicationOpenedByNotification(@Nullable final String notificationI
       sessionManager.onDirectInfluenceFromNotificationOpen(appEntryState, notificationId);
    }
 
-   static boolean startOrResumeApp(Activity inContext) {
-      Intent launchIntent = inContext.getPackageManager().getLaunchIntentForPackage(inContext.getPackageName());
-      logger.debug("startOrResumeApp from context: " + inContext + " isRoot: " + inContext.isTaskRoot() + " with launchIntent: " + launchIntent);
-      // Make sure we have a launcher intent.
-      if (launchIntent != null) {
-         if (inContext.isTaskRoot()) {
-            inContext.startActivity(launchIntent);
-         } else {
-            launchIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-            PendingIntent.getActivity(inContext, 0, launchIntent, 0);
-         }
-         return true;
-      }
-      return false;
+   static boolean startOrResumeApp(Activity activity) {
+      Intent launchIntent = activity.getPackageManager().getLaunchIntentForPackage(activity.getPackageName());
+      logger.debug("startOrResumeApp from context: " + activity + " isRoot: " + activity.isTaskRoot() + " with launchIntent: " + launchIntent);
+
+      // Not all apps have a launcher intent, such one that only provides a homescreen widget
+      if (launchIntent == null)
```

Comment:
How does this prevent the application from being launched from the launching activity when the application is in the background, instead of resuming the activity on top of the stack?
Doesn't this remove the behavior of emulating app icon click when the app is in the background when clicking a notification?

Reply:
This bug was we were missing a `startActivity` in the else of the `isTaskRoot` check. This was fixed in commit 0326d93abb354aa3569e945ba6365aeab9dcd9d0 of this PR.

I noticed `getLaunchIntentForPackage` defaults to `FLAG_ACTIVITY_NEW_TASK` so the whole `isTaskRoot` check and setting of it isn't needed. Notes to this and ASOP source linked of this in commit b27ac140eb5e74bfe55468ea4de9426b7bff2a3a

I didn't notice any other behavior differences before and after this PR. See the manual test scenarios above for what I tested.

This PR is scoped to fixing this one specific cold start bug. However I did notice a resume issue which I will be making another PR based on this one to fix that as well. We can follow up there on that.